tests/unit/auth_adapter/test_get_session.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the dependency for geting the current session."""
+
+from __future__ import annotations
+
+from fastapi import HTTPException, Request, status
+from pytest import mark, raises
+
+from auth_service.auth_adapter.adapters.memory_session_store import MemorySessionStore
+from auth_service.auth_adapter.deps import get_session
+from auth_service.config import CONFIG
+
+SESSION_COOKIE = "session"
+CSRF_TOKEN_HEADER = "X-CSRF-Token"
+
+SESSION_ARGS = {
+    "ext_id": "john@aai.org",
+    "user_name": "John Doe",
+    "user_email": "john@home.org",
+}
+
+UNCRITICAL_METHODS = ["GET", "HEAD", "OPTION"]
+CRITICAL_METHODS = ["POST", "PUT", "PATCH", "DELETE"]
+ALL_METHODS = UNCRITICAL_METHODS + CRITICAL_METHODS
+
+EXPECTED_ERROR_STATUS = status.HTTP_401_UNAUTHORIZED
+EXPECTED_ERROR_MESSAGE = "Invalid or missing CSRF token"
+
+
+async def assert_get_session(
+    method: str = "GET",
+    with_cookie: bool | str = True,
+    with_csrf_token: bool | str = True,
+    expect_result: bool | None = True,
+) -> None:
+    """Assert that the session is retrieved correctly.
+
+    If `expect_result` is None, then the session should not be found.
+    If it is False, then HTTPException with a CSRF error should be raised.
+    """
+    store = MemorySessionStore(config=CONFIG)
+
+    session = await store.create_session(**SESSION_ARGS)
+
+    if with_cookie is False:
+        cookie = None
+    elif with_cookie is True:
+        cookie = session.session_id
+    else:
+        cookie = None
```

Comment:
```suggestion
    if with_cookie is True:
        cookie = session.session_id
    else:
        cookie = None
```
This can be simplified

Reply:
Thanks for catching this, it was actually wrong. Fixed in a9020ff.